### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Afollet**
Refactored Tests and DataImportService

Removed GetPatient from DataImportService. Replaced with a new patient creation in the
dependent methods. Placed thrown exception for null or blank externalID in getPatienSnapshots
and createPatient. Created Tests for TreatmentProtocol, DataImortService, and a timed test
for creating and finding DataProjection Nodes.

### CODE REVIEW

File: data-model/src/test/java/org/pdxfinder/BaseTestWithPersistence.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
```
Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/BaseTestWithPersistence.java]: This comment is unnecessary

File: data-model/src/main/java/org/pdxfinder/graph/dao/TreatmentProtocol.java [hunk 2]
```diff
@@ -110,7 +110,7 @@ public String getTreatmentString(boolean includeControlDrugs){
 
             String drugName;
 
-            if(comp.getTreatment() != null && comp.getTreatment().getTreatmentToOntologyRelationship() != null & comp.getTreatment().getTreatmentToOntologyRelationship().getOntologyTerm() != null){
+            if(comp.getTreatment() != null && comp.getTreatment().getTreatmentToOntologyRelationship() != null && comp.getTreatment().getTreatmentToOntologyRelationship().getOntologyTerm() != null){
```
Comment (rmwthorne) [inline data-model/src/main/java/org/pdxfinder/graph/dao/TreatmentProtocol.java]: You might want to split this across multiple lines for clarity

File: data-model/src/test/java/org/pdxfinder/TestConfigWithPersistence.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+@EnableTransactionManagement
+@ComponentScan(value = "org.pdxfinder")
+@EnableNeo4jRepositories("org.pdxfinder.graph.repositories")
+public class TestConfigWithPersistence {
+
+    @Bean
+    public org.neo4j.ogm.config.Configuration getConfiguration() {
+        org.neo4j.ogm.config.Configuration config = new org.neo4j.ogm.config.Configuration();
+
+        //   To persist the database, uncomment this section
```
Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/TestConfigWithPersistence.java]: This section should be factored out if possible

Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/TestConfigWithPersistence.java]: You can delete this unnecessary comment

File: data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        TreatmentProtocol treatmentProtocol = new TreatmentProtocol();
+        treatmentString3 = treatmentProtocol.getTreatmentString(true);
+
+        Assert.assertEquals("", treatmentString1);
+        Assert.assertEquals("", treatmentString2);
+        Assert.assertEquals("", treatmentString3);
+    }
+
+    @Test
+    public void Given_getDostString_When_ControlFlagIsToggled_Then_returnAppropriateDoseString() {
```
Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java]: Minor typo here
```suggestion
    public void Given_getDoseString_When_ControlFlagIsToggled_Then_returnAppropriateDoseString() {
```

File: data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        treatmentProtocol2.setComponents(componentsList2);
+
+        treatmentProtocol.setResponse(testResponse);
+        treatmentProtocol2.setResponse(testResponse2);
+
+        return Arrays.asList(treatmentProtocol,treatmentProtocol2);
+    }
+
+    @Test
+    public void Given_getTreatmentString_When_MultipleTCandControlsOn_Then_ReturnNotNullConcatenatedString(){
```
Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java]: Needs to be renamed so naming convention is consistent - Compare `_getTreatment` and `_MultipleTC`. Same issue in other test names

File: data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+                .getTreatmentString(READ_CONTROLS_COMPONENTS);
+
+        Assert.assertNotNull(treatmentString);
+        Assert.assertEquals(
+                String.format("%s and %s and %s", CONTROL_ONTOLOGY_LABEL, DRUG_ONTOLOGY_LABEL, DRUG_ONTOLOGY_LABEL_2),
+                treatmentString);
+    }
+
+    @Test
+    public void Given_getTreatmentString_WhenTreatmentComponentsWithNoControl_Then_returnConcatenatedStrings(){
```
Comment (rmwthorne) [inline data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java]: `When` needs a following underscore
```suggestion
    public void Given_getTreatmentString_When_TreatmentComponentsWithNoControl_Then_returnConcatenatedStrings(){
```

File: data-services/src/main/java/org/pdxfinder/services/DataImportService.java [hunk 7]
```diff
@@ -396,13 +401,19 @@ public Patient findPatient(String patientId, Group dataSource){
 
     public PatientSnapshot getPatientSnapshot(String externalId, String sex, String race, String ethnicity, String age, Group group) {
 
+
+        if(externalId == null || externalId.equals("")){
+            log.warn("In DataImportService.createPatient() : patientId is null");
```
Comment (rmwthorne) [inline data-services/src/main/java/org/pdxfinder/services/DataImportService.java]:
```suggestion
            log.warn("In DataImportService.createPatient() : patientId is null or blank");
```

File: data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+         actualSnapshot = dataImportService.getPatientSnapshot(EXTERNAL_ID, SEX, RACE,
+                 ETHNICITY, AGE_AT_COLLECTION, GROUP);
+
+        Assert.assertEquals(EXTERNAL_ID,  actualSnapshot.getPatient().getExternalId());
+        Assert.assertEquals(AGE_AT_COLLECTION,  actualSnapshot.getAgeAtCollection());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void Given_getpatiensnapshot6param_When_externalidisblank_Then_returnnullexception(){
```
Comment (rmwthorne) [inline data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java]: change test name to match others, same with the below test. Minor  typo here too.
```suggestion
    public void Given_GetPatientSnapshot6Param_When_ExternalIdIsBlank_Then_ReturnNullException(){
```

File: data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+        actualSnapshot = dataImportService.getPatientSnapshot("", SEX, RACE,ETHNICITY,
+                AGE_AT_COLLECTION, GROUP);
+
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void Given_getpatiensnapshot6param_When_externalidisnull_Then_returnnullexception(){
```
Comment (rmwthorne) [inline data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java]: As above comment 
```suggestion
    public void Given_GetPatientSnapshot6Param_When_ExternalIdIsNull_Then_ReturnNullException(){
```

File: data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+        HashSet<PatientSnapshot> pSnaps = new HashSet<>();
+        pSnaps.add(expectedSnapshot);
+
+        when(this.patientSnapshotRepository.findByPatient(EXTERNAL_ID))
+                .thenReturn(pSnaps);
+
+        actualSnapshot = dataImportService.getPatientSnapshot(PATIENT, AGE_AT_COLLECTION);
+
+        Assert.assertEquals(expectedSnapshot, expectedSnapshot);
```
Comment (CsabaHalmagyi) [inline data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java]: Compare expectedSnapshot with actualSnapshot here.

**Comment by Afollet** [comment]
> style: avoid wildcard imports

>Looking into it, I don't think you can avoid it here?

Yes, it can be done:

Import static org.mockito.Mockito.when;

**Pull request merged**